File: backend/src/main/java/com/deliverXY/backend/NewCode/wallet/service/WalletService.java
package com.deliverXY.backend.NewCode.wallet.service;

import com.deliverXY.backend.NewCode.common.enums.PaymentProvider;
import com.deliverXY.backend.NewCode.user.domain.AppUser;
import com.deliverXY.backend.NewCode.wallet.domain.TopUpRequest;
import com.deliverXY.backend.NewCode.wallet.domain.Wallet;
import com.deliverXY.backend.NewCode.wallet.dto.WalletTransactionDTO;

import java.math.BigDecimal;
import java.util.List;

public interface WalletService {
    Wallet getWallet(Long userId);
    void createWalletForUser(AppUser user);
    TopUpRequest initiateTopUp(Long userId, BigDecimal amount, PaymentProvider provider);
    void finalizeTopUp(Long topUpId, boolean success, String referenceId);
    void deposit(Long userId, BigDecimal amount, String reference);

    boolean withdraw(Long userId, BigDecimal amount, String reference);

    void addTransaction(Long userId, BigDecimal amount, String type, String reference);

    List<WalletTransactionDTO> getTransactions(Long userId);
}

File: frontend/src/services/wallet.service.ts
import { apiRequest } from "./api";


export async function getWallet(){
    return await apiRequest("GET", "/api/wallet");
}
export async function getWalletTransactions() {
    return apiRequest("GET", "/api/wallet/transactions");
  }
  

  export async function initiateTopUp(
    amount: number,
    provider: "STRIPE" | "MOCK" = "STRIPE"
  ) {
    return apiRequest("POST", "/api/wallet/topup/initiate", {
      amount,
      provider,
    });
  }
  export async function finishTopUp(
    topUpId: number,
    success: boolean,
    referenceId?: string
  ) {
    return apiRequest(
      "POST",
      `/api/wallet/topup/callback?topUpId=${topUpId}&success=${success}&referenceId=${referenceId ?? ""}`
    );
  }
  
  export async function withdrawFromWallet(
    amount: number,
    reference?: string
  ) {
    return apiRequest("POST", "/api/wallet/withdraw", {
      amount,
      reference: reference ?? "USER_WITHDRAW",
    });
  }
  
  export async function depositToWallet(
    amount: number,
    reference?: string
  ) {
    return apiRequest("POST", "/api/wallet/deposit", {
      amount,
      reference: reference ?? "MANUAL_DEPOSIT",
    });
  }